### PR TITLE
Return timestamp, source, level and message in GitOps Run log endpoint

### LINKED ISSUE #3280: Return timestamp, source, level and message in GitOps Run log endpoint
Currently, the get-session-logs endpoint returns logs as an array of strings, which is not enough to be used by the front end.

We should return logs as an array of tuples, each of which contains
- timestamp (standard logging format)
- source (the source of the log - it would be "gitops-run-client" to indicate that the log is from the GitOps Run console, or "$namespace/$resource" to indicate that it's from pods of the resource (e.g. deployment, statefuleset, daemonset, etc)
- level (the logs level) they are 3 levels, "info", "warning" and "error".
- message (the log itself)

### PR DESCRIPTION
**Pull request by opudrovs**
Closes #3280

- Added the `LogEntry` structure to the `GetSessionLogsResponse` type.

- Ran `make proto`.

- Added collecting data for `Log` structure fields.

- Added marshaling session log data to JSON.

- Added unmarshaling log entries from JSON in the `GetSessionLogs` endpoint.

- Updated integration tests for `GetSessionLogs`.

Testing:
I just updated integration tests for the `GetSessionLogs` endpoint for now. Unable to test it in the UI because of https://github.com/weaveworks/weave-gitops/issues/3313 But for the scope of this ticket automated tests should work fine.

### CODE REVIEW

File: pkg/logger/s3_log_writer.go [hunk 1]
```diff
@@ -36,13 +40,38 @@ func CreateBucket(minioClient *minio.Client, bucketName string) error {
...
+	if len(msg) > 0 {
+		if strings.HasPrefix(msg, "✗ ") {
+			level = "error"
+		} else if strings.HasPrefix(msg, "⚠️ ") {
+			level = "warning"
+		}
+	}
+
+	result := &pb.LogEntry{
+		Timestamp: now.Format(time.RFC3339),
```
Comment (chanwit) [inline pkg/logger/s3_log_writer.go:56]: :+1:

File: pkg/logger/s3_log_writer.go [hunk 2]
```diff
@@ -36,13 +40,38 @@ func CreateBucket(minioClient *minio.Client, bucketName string) error {
 }
 
 func (l *S3LogWriter) putLog(msg string) {
+	now := time.Now()
+
+	level := "info"
+
+	if len(msg) > 0 {
+		if strings.HasPrefix(msg, "✗ ") {
```
Comment (chanwit) [inline pkg/logger/s3_log_writer.go:48]:
```suggestion
		if strings.HasPrefix(msg, "✗") {
```
without the space would be just fine :-)

File: pkg/logger/s3_log_writer.go [hunk 3]
```diff
@@ -18,6 +21,7 @@ type S3LogWriter struct {
 
 const SessionLogBucketName = "gitops-run-logs"
 const PodLogBucketName = "pod-logs"
+const SessionLogSource = "gitops-run-client"
```
Comment (chanwit) [inline pkg/logger/s3_log_writer.go:24]: :+1:

**Review by chanwit — COMMENTED** [review]
LGTM in general!

Will do the second round when it's ready.

File: pkg/logger/s3_log_writer.go (hunk 2 above)
Comment (opudrovs) [reply pkg/logger/s3_log_writer.go]: Removed the space :)

Comment (opudrovs) [reply pkg/logger/s3_log_writer.go]: And in the other place too.

File: pkg/logger/s3_log_writer.go (hunk 1 above)
Comment (opudrovs) [reply pkg/logger/s3_log_writer.go:56]: I wasn't sure which timestamp format you need for this endpoint, so copied it from the timestamp field in another endpoint. Glad it is OK.

**Review by chanwit — APPROVED** [review]

**Pull request merged**